### PR TITLE
refactor: put pur before eff in AST nodes

### LINKED ISSUE #3766: Purity should go before effects in ASTs
I am merging this, but I think purity should go before effects. Will you create a separate PR?

_Originally posted by @magnus-madsen in https://github.com/flix/flix/issues/3730#issuecomment-1127467458_

### PR DESCRIPTION
**Pull request by mlutze**
fixes #3766

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala [hunk 1]
```diff
@@ -2137,18 +2137,18 @@ object ParsedAst {
...
       * Represents an effect implemented as a set.
       */
-    case class Effect(s: EffectSet) extends EffectOrPurity
+    case class Effect(s: EffectSet) extends PurityOrEffect
 
     /**
       * Represents an effect implemented as a boolean formula.
       */
-    case class Purity(b: Type) extends EffectOrPurity
+    case class Purity(b: Type) extends PurityOrEffect
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala:2151]: I guess this should come first also :D

File: main/test/flix/CompilerSuite.scala [hunk 2]
```diff
@@ -142,7 +142,8 @@ class CompilerSuite extends FlixSuite(incremental = true) {
   //
   // Effects
   //
-  mkTest("main/test/flix/Test.Exp.Effect.flix")
+  // TODO temporarily allowing redundancies during development
+  mkTest("main/test/flix/Test.Exp.Effect.flix")(options.copy(xallowredundancies = true))
```
Comment (magnus-madsen) [inline main/test/flix/CompilerSuite.scala:146]: Nina will soon come after you :)

**Comment by magnus-madsen** [comment]
Thanks!

File: main/test/flix/CompilerSuite.scala (hunk 2 above)
Comment (mlutze) [reply main/test/flix/CompilerSuite.scala:146]: Guess I better hurry with the development!

**Pull request merged**